doc/TRANSACTION.md

```diff
@@ -1,56 +1,62 @@
-# BOLOK Transaction Serialization
+# Hathor Transaction
 
 ## Overview
 
-The custom transaction serialization presented is for the purely fictitious BOLOK *chain* which has been inspired by other popular blockchain (see [Links](#links)).
+The hathor transaction serialized
 
-## Amount units
+## Structure
 
-The base unit in BOLOK *chain* is the BOL and the smallest unit used in raw transaction is the *bolino* or mBOL: 1 BOL = 1000 mBOL.
+### Transaction
 
-## Address format
+| Field | Size (bytes) | Description |
+| --- | :---: | --- |
+| `change info` | var | Bytes representing the existence of a change output and the bip32 path of the change address |
+| `tx_version` | 2 | TX Version, indicates this is a transaction, block, etc. |
+| `num_tokens` | 1 | Number of token UIDs present on the data below |
+| `num_inputs` | 1 | Number of inputs present on the data below |
+| `num_outputs` | 1 | Number of outputs present on the data below |
+| `tokens` | 32\*`num_tokens` | list of Token UIDs |
+| `inputs` | 35\*`num_inputs` | list of Inputs |
+| `outputs` | var | List of Outputs |
 
-BOLOK addresses are hexadecimal numbers, identifiers derived from the last 20 bytes of the Keccak-256 hash of the public key.
+#### Change info
 
-## Structure
+| Field | Size | Description |
+| --- | --- | --- |
+| `has_change_output` | 1bit | Indicates existence of a change output.<br>(mask `0x80`)<br>If false, the whole first byte is 0. |
+| `change bip32 path length` | 4bits | if `has_change_output=true` this will be the last 4 bits of the first byte<br>(mask `0x0F`) |
+| `change output index` | 1 byte | the index of the change output |
+| `change bip32 path` | var (`4*bip32_path_length`) | an array of 32bit unsigned integers of the path indexes |
 
-### Transaction
+Obs: bip32 paths have a hard depth limitation of 5, so the max value of the change info is 22 bytes
```

Comment:
Do we enforce this limitation in our code?

Reply:
Yes, on `src/common/bip32.c` we test the length with `MAX_BIP32_PATH` defined on `src/common/bip32.h` as 5

Reply:
We can allow longer paths (length up to 10 is the usual), but since we don't use derivations past this I think we shouldn't.